### PR TITLE
Purge pagination

### PR DESCRIPTION
**Pull request by sejoker**
Second try to fix https://github.com/cloudflare/Cloudflare-WordPress/issues/337. In addition to previous https://github.com/cloudflare/Cloudflare-WordPress/pull/339/files I limit the number of pages to 3.
Also APO applies Edge Cache TTL of 30 mins for all `example.com/page/*` paginated urls.

### CODE REVIEW

File: src/WordPress/Hooks.php [hunk 1]
```diff
@@ -256,6 +256,20 @@ public function getPostRelatedLinks($postId)
             array_push($listofurls, $pageLink);
         }
 
+        // Refresh pagination
+        $total_posts_count = wp_count_posts()->publish;
+        $posts_per_page = get_option('posts_per_page');
+        // Limit to up to 3 pages
+        $page_number_max = min(3, ceil($total_posts_count / $posts_per_page));
```
Comment (xtuc) [inline src/WordPress/Hooks.php:263]:
```suggestion
        $page_number_max = max(3, ceil($total_posts_count / $posts_per_page));
```

**Review by xtuc — APPROVED** [review]

File: src/WordPress/Hooks.php (hunk 1 above)
Comment (jacobbednarz) [inline src/WordPress/Hooks.php:263]: i wonder if we should make this configurable for those who want to do say 2 or 10 instead of the static 3?

**Review by jacobbednarz — APPROVED** [review]
A non-blocking question, otherwise 🚢

File: src/WordPress/Hooks.php [hunk 2]
```diff
@@ -256,6 +256,20 @@ public function getPostRelatedLinks($postId)
...
+
+        $this->logger->debug("total_posts_count $total_posts_count");
```
Comment (jacobbednarz) [inline src/WordPress/Hooks.php:265]: do we need to keep these debug lines around?

Comment (sejoker) [reply src/WordPress/Hooks.php:265]: > do we need to keep these debug lines around?

I think yes, we don't enable logging by default, so when we need logging it's useful to include.

File: src/WordPress/Hooks.php (hunk 1 above)
Comment (sejoker) [reply src/WordPress/Hooks.php:263]: We can do this once requested by customers. I think it's ok to ship without configuration option.

**Pull request merged**